src/python/ensembl/io/genomio/fasta/chunk.py

```diff
@@ -14,48 +14,99 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+""""Split a set of nucleotide sequence(s) (.fasta, .gz) into smaller chunks."""
 
+__all__ = [
+    "split_seq_by_chunk_size",
+    "split_seq_by_n",
+]
 
-import argparse
-import gzip
+import logging
 import os
-import sys
 import re
+from typing import Optional, Pattern
 
 from Bio import SeqIO
 from Bio.Seq import Seq
 from Bio.SeqRecord import SeqRecord
 
 from ensembl.utils.archive import open_gz_file
+from ensembl.utils.argparse import ArgumentParser
+from ensembl.utils.logging import init_logging_with_args
 
 
-def get_args():
-    parser = argparse.ArgumentParser()
-    # in files
-    parser.add_argument(
-        "fasta_dna",
-        metavar="input.fna[.gz]",
-        type=str,
-        help="fasta (possibly gzipped) file with dna sequences to split",
+def split_seq_by_n(seq: str, split_patten: Pattern) -> list:
+    """Split a string into chunks at the positions of N sequences.
+    (return [ len(seq) ] if no split_patten)
+
+    Args:
+        seq: Sequence to be split into chunks.
+        split_patten: Split points of sequence chunks.
+    """
+    seq_len = len(seq)
+    if not split_patten:
+        return [seq_len]
+    split_points = [m.end() for m in split_patten.finditer(seq)]
+    if not split_points or split_points[-1] != seq_len:
+        split_points.append(seq_len)
+    return split_points
+
+
+def split_seq_by_chunk_size(
+    ends: list[int], chunk_size: int, tolerated_chunk_len: Optional[int] = None
+) -> list:
+    """Split list of end coordinates, to form chunks not longer then
+    chunk_size (tolerated_chunk_len if specified).
+
+    Args:
+        ends: List of one or more chunk(s) to split a sequence.
+        chunk_size: Size of chunk to be split sequence.
+        tolerated_chunk_len: Constrained chunk size when splitting a sequence.
+    """
+    if tolerated_chunk_len is None or tolerated_chunk_len < chunk_size:
+        tolerated_chunk_len = chunk_size
+    result = []
+    offset = 0
+    for chunk_end in ends:
+        chunk_len = chunk_end - offset
+        if chunk_len > tolerated_chunk_len:
+            # exclude starts, as they are 0 or pushed as previous chunk_ends
+            result += list(range(offset, chunk_end, chunk_size))[1:]
+        result.append(chunk_end)
+        offset = chunk_end
+    return result
+
+
+## MAIN ##
+def main():
+    """Modules entry-point."""
```

Comment:
```suggestion
    """Module's entry-point."""
```

Reply:
I believe this suggestion isn't correct. The plural of module modules. Module's == module is for short. I just will remove the s entirely